src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -54,19 +55,23 @@ const defaultProps = {
 };
 
 function _needsReProjectPoints(oldProps, props) {
-  return oldProps.radius !== props.radius;
+  return oldProps.radius !== props.radius || oldProps.hexagonAggregator !== props.hexagonAggregator;
 }
 
 function _percentileChanged(oldProps, props) {
   return oldProps.lowerPercentile !== props.lowerPercentile ||
     oldProps.upperPercentile !== props.upperPercentile;
 }
 
+function _needsReSortBins(oldProps, props) {
+  return oldProps.colorValueAccessor !== props.colorValueAccessor;
+}
+
 export default class HexagonLayer extends Layer {
   constructor(props) {
-    if (!props.radius) {
-      log.once(0, 'HexagonLayer: radius in meter is needed to aggregate points into ' +
-        'hexagonal bins, Now using 1000 meter as default');
+    if (!props.radius || !typeof props.hexagonAggregator === 'function') {
```

Comment:
Is this check for hexagonAggregator needed when we have a default? Is it intended to cover the user putting in some unexpected value in this prop?

I know I have said this before, but we should implement a `propTypes` system to avoid adding ad-hoc checks like this.